tests/neg-custom-args/isInstanceOf/Result.scala

```diff
@@ -0,0 +1,11 @@
+object p {
+
+  // test parametric case classes, which synthesis `canEqual` and `equals`
+  enum Result[+T, +E] {
+    case OK [T](x: T) extends Result[T, Nothing]
+    case Err[E](e: E) extends Result[Nothing, E]
```

Comment:
What is tested here? There is no pattern match

Reply:
Here the synthesized code will generate pattern match in `canEqual` and `equals`.